woocommerce/woocommerce
# Create new component TreeSelect and migrate category field

### All Submissions:






### Changes proposed in this Pull Request:



Some time ago the Tree control was written and we decided to migrate the category field to it. 

Previously, the Select control was used for the input, with a callback to render the children as a popover. I had a lot of problems trying to make Select work with TreeControl, so I decided to create a new component to wrap this complexity and also stopped using SelectControl.

Closes #37043.
Closes #35969. 



### How to test the changes in this Pull Request:




1. Run `pnpm install` and `pnpm build`
2. Open `/wp-admin/admin.php?page=wc-admin&path=%2Fadd-product-old`
3. Go to the Category field.
4. Test general usage, both with keyboard and mouse.
5. Compare to the `trunk` version. Check if you see any regressions.



### Other information:




### FOR PR REVIEWER ONLY:
